lib/src/hydrated.dart

```diff
@@ -163,72 +138,47 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
   @override
   StackTrace? get stackTrace => _subject.stackTrace;
 
+  bool get _doEncodePersistedValue => _hydrate != null && _persist != null;
+
   /// Hydrates the HydratedSubject with a value stored on the user's device.
   ///
   /// Must be called to retrieve values stored on the device.
   Future<void> _hydrateSubject() async {
-    final prefs = await SharedPreferences.getInstance();
-
-    T? val;
+    try {
+      T? val;
+      if (_doEncodePersistedValue) {
```

Comment:
```suggestion
      if (_hydrate != null) {
```
maybe that would be clearer?

Reply:
Agreed. Suggestion applied manually.